packages/ui/src/components/Visualization/Canvas/Canvas.tsx

```diff
@@ -137,6 +139,24 @@ export const Canvas: FunctionComponent<PropsWithChildren<CanvasProps>> = ({ enti
     }
   }, [selectedIds, controller]);
 
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Delete' && selectedNode) {
+        const { canRemoveStep, canRemoveFlow } = selectedNode.data!.vizNode!.getNodeInteraction();
+        if (!canRemoveStep && !canRemoveFlow) {
+          return;
+        }
+
+        onDeleteStep();
+        setSelectedNode(undefined);
+        setSelectedIds([]);
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  }, [selectedNode, entities, onDeleteStep]);
```

Comment:
Maybe it would be better to use an `onKeyDown` property in one of the `CustomNode` elements, so we don't need to imperatively assign an event listener.